include/seqan3/alphabet/concept.hpp

```diff
@@ -47,42 +47,12 @@ namespace seqan3
 {
 
 // ------------------------------------------------------------------
-// free functions to member function wrapper
-// ------------------------------------------------------------------
-// move to alphabet_detail.hpp?
-
-
-/* The public alphabet concept requires only free function access
- * for type that have member functions we create a wrapper here
- * so you don't have to repeat it.
- */
-//! \privatesection
-namespace detail
-{
-
-template <typename t>
-concept bool internal_alphabet_concept = requires (t t1)
-{
-    typename t::char_type;
-    typename t::integral_type;
-    t::value_size;
-
-    { t1.to_char() } -> typename t::char_type;
-    { t1.to_integral() } -> typename t::integral_type;
-
-    { t1.from_char('a')   } -> t;
-    { t1.from_integral(0) } -> t;
-};
-} // namespace seqan3::detail
-//! \publicsection
-
-// ------------------------------------------------------------------
```

Comment:
Our real alphabet concept expects free functions, but for our own alphabet implementations we only provide members, because it is easier/cleaner to write, and people might want to use members, too. So for our alphabets to satisfy our own concept, we need to free function -> member wrappers.
Previously I defined a "internal_alphabet" to constrain these free functions. But it caused lots of confusion. Also it is unclear whether to put it into `detail`, than it is not clear to users how the free functions are constrained. Or to put it into `seqan3` than it adds more confusion for users what's an` internal alphabet concept`?
Another benefit of having individual constraints instead of an extra concept is, that some types can combine local/global overloads and it will work.